specification/sections/03-evaluation-context.md

```diff
@@ -128,4 +133,79 @@ The SDK implementation must run the `on context changed` handler on all register
+
+[![experimental](https://img.shields.io/static/v1?label=Status&message=experimental&color=orange)](https://github.com/open-feature/spec/tree/main/specification#experimental)
+
+#### Condition 3.3.1
+
+> The implementation uses the dynamic-context paradigm.
+
+see: [dynamic-context paradigm](../glossary.md#dynamic-context-paradigm)
+
+##### Conditional Requirement 3.3.1.1
+
+> The API **MUST** have a method for setting a `transaction context propagator`.
+
+If there already is a `transaction context propagator`, it is replaced with the new one. 
+
+##### Conditional Requirement 3.3.1.2
+
+> The API **MUST** have a method for setting the `evaluation context` of the `transaction context propagator`.
+
+If a `transaction context propagator` is set, this `evaluation context` must be supplied to it and so will be available during the current transaction.
+If no `transaction context propagator` is set, this `evaluation context` is not used for evaluations.
+This method can e.g. be used in a request handler to add request-specific information to the `evaluation context`.
+
+##### Conditional Requirement 3.3.1.3
+
+> A `transaction context propagator` **MUST** have a method for setting the `evaluation context` of the current transaction.
+
+A `transaction context propagator` is responsible for persisting context for the duration of a single transaction.
+Typically, a transaction context propagator will propagate the context using a language-specific carrier such as [ThreadLocal (Java)](https://docs.oracle.com/javase/8/docs/api/java/lang/ThreadLocal.html), [async hooks (Node.js)](https://nodejs.org/api/async_hooks.html), [Context (Go)](https://pkg.go.dev/context) or another similar mechanism.
+
+The following shows a possible TypeScript implementation using [AsyncLocalStorage (async_hooks)](https://nodejs.org/api/async_context.html): 
+
+```typescript
+export class AsyncLocalStorageTransactionContext implements TransactionContextPropagator {
+    private asyncLocalStorage = new AsyncLocalStorage<EvaluationContext>();
+
+    getTransactionContext(): EvaluationContext {
+        return this.asyncLocalStorage.getStore() ?? {};
+    }
+    setTransactionContext(context: EvaluationContext, callback: () => void): void {
+        this.asyncLocalStorage.run(context, callback);
+    }
+}
+
+/**
+ * This example is based on an express middleware.
+ */
+app.use((req: Request, res: Response, next: NextFunction) => {
+    const ip = res.headers.get("X-Forwarded-For")
+    OpenFeature.setTransactionContext({ targetingKey: req.user.id, ipAddress: ip }, () => {
+        // The transaction context is used in any flag evaluation throughout the whole call chain of next 
+        next();
+    });
+})
+```
+
+##### Conditional Requirement 3.3.1.4
+
+> A `transaction context propagator` **MUST** have a method for getting the `evaluation context` of the current transaction.
+
+This must be used by the SDK implementation when merging the context for evaluating a feature flag.
```

Comment:
```suggestion
This will be used by the SDK implementation when merging the context for evaluating a feature flag.
```

Try to avoid RFC2119 keywords in non-normative sections, where possible.

I would add an equivalent sentence in 3.3.1.3, something like this "This must be used by the SDK when setting the context for a transaction"